### PR TITLE
Fix the generation of nix3 manpages

### LINKED ISSUE #5051: nix3 manpages no longer get installed
**Describe the bug**

Since https://github.com/NixOS/nix/commit/c5f5d615a66c2b9411857914893eb854796a482f, nix3 manpages seem to not get installed anymore. 

**Steps To Reproduce**

Run the following and notice no `nix3-*` manpages are there.
```
nix build 'github:nixos/nix?rev=c5f5d615a66c2b9411857914893eb854796a482f'
ls result/share/man/man1
```

**Expected behavior**

Using the preceding commit as a point of reference, the nix3 manpages are installed.
```
nix build 'github:nixos/nix?rev=db4d4cf4ba42727edd8cc5a6c516554fb8a6b688'
ls result/share/man/man1
```
**`nix-env --version` output**
```
nix-env (Nix) 2.4pre20210719_c5f5d61
```

### PR DESCRIPTION
**Pull request by regnat**
- Separate the generation of the manpages from their installation
- Make sure that `make` generates the manpages
- Make sure that `make install` installs them

Fix #5051

### CODE REVIEW

File: doc/manual/local.mk [hunk 1]
```diff
@@ -77,21 +73,29 @@ $(d)/builtins.json: $(bindir)/nix
...
+
+$(mandir)/man1/nix3-manpages: doc/manual/generated/man1/nix3-build.1
+	@mkdir -p $$(dirname $@)
+	$(trace-install) install -m 0644 $$(dirname $<)/* $$(dirname $@)
 
 # Technically this rule generates all the `nix3-*` manpages, but since we don’t
 # know their list statically and they are all generated at once anyways, we can
 # just be dirty and only track one
-$(mandir)/man1/nix3-build.1: $(d)/src/command-ref/new-cli
+doc/manual/generated/man1/nix3-manpages: $(d)/src/command-ref/new-cli
```
Comment (alyssais) [inline doc/manual/local.mk:87]: The comment above is now out of date.

**Comment by edolstra** [comment]
Looks like `make install` no longer works unless you do `make` first:
```
$ rm -rf doc/manual/generated/

$ make install -j16
make: *** No rule to make target 'doc/manual/generated/man1/nix3-build.1', needed by '/home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-manpages'.  Stop.
```

**Comment by regnat** [comment]
> Looks like `make install` no longer works unless you do `make` first:

Duh my bad, I misnamed a dependency. Should be fixed now

File: doc/manual/local.mk (hunk 1 above)
Comment (regnat) [reply doc/manual/local.mk:84]: Indeed, thanks

**Pull request merged**